core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -78,6 +79,14 @@
      */
     RequestContext newDerivedContext(Request request);
 
+    default ContentPreviewerFactory requestContentPreviewerFactory() {
```

Comment:
How about removing these properties and using `ctx.options().request/responseContentPreviewerFactory()`, given that these properties are not likely to be accessed by users?

Reply:
`ClientRequestContext` has `options()` but `ServiceRequestContext` doesn't have things like that so I need to use 'virtualHost().requestContentPreviewerFactory()' instead. It means that specified castings will be needed in `DefaultRequestLog.requestHeaders()` and `DefaultRequestLog.responseHeaders()` What do you think?

Reply:
How about adding constructor parameters to `DefaultRequestLog` so that you can pass them in the constructor of `DefaultClient/ServiceRequestContext`?